src/databricks/labs/remorph/coverage/coverage_utils.py

```diff
@@ -0,0 +1,173 @@
+import dataclasses
+import json
+import logging
+import os
+import subprocess
+from collections.abc import Generator
+from datetime import datetime
+from pathlib import Path
+from typing import TextIO
+
+import sqlglot
+from sqlglot import Expression
+from sqlglot.dialects.dialect import Dialect
+from sqlglot.errors import ErrorLevel
+
+logger = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class ReportEntry:
+    project: str
+    commit_hash: str | None
+    version: str | None
```

Comment:
```suggestion
    version: str
```

Reply:
Use https://github.com/databrickslabs/blueprint?tab=readme-ov-file#unreleased-version-detection to get unreleased version